Philips.CodeAnalysis.MaintainabilityAnalyzers/AvoidAssignmentInConditionAnalyzer.cs

```diff
@@ -0,0 +1,95 @@
+﻿// © 2020 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers
+{
+	/// <summary>
+	/// Diagnostic for variable assignment inside an if condition.
+	/// </summary>
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidAssignmentInConditionAnalyzer : DiagnosticAnalyzer
+	{
+		private const string Title = "Assignment in condition.";
+		private const string Message = "Assignment within condition on line {0}.";
+		private const string Description = "Assignment in condition.";
+		private const string Category = "Usage";
+
+		private static readonly DiagnosticDescriptor Rule =
+			new DiagnosticDescriptor(
+				Helper.ToDiagnosticId(DiagnosticIds.AvoidAssignmentInCondition),
+				Title,
+				Message,
+				Category,
+				DiagnosticSeverity.Error,
+				isEnabledByDefault: true,
+				description: Description
+			);
+
+		/// <summary>
+		/// <inheritdoc cref="DiagnosticAnalyzer"/>
+		/// </summary>
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Rule);
+
+		/// <summary>
+		/// <inheritdoc cref="DiagnosticAnalyzer"/>
+		/// </summary>
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```

Comment:
Do you need to check generated code?  What is the user experience when generated code triggers this?  Can they fix it?

Reply:
Your right, removed it from generated code